internal/parse/parser.go

```diff
@@ -243,58 +236,64 @@ func (p *Parser) skipName() bool {
 //  - bool == false, err == nil
 //		The construct was not the one we are looking for
 
-// parseIdentifier parses either a name made up only of nameBytes or an
-// asterisk.
-func (p *Parser) parseIdentifier() (string, bool) {
-	if p.skipByte('*') {
-		return "*", true
-	}
+type asteriskFlag int
+
+const (
+	allowAsterisk asteriskFlag = iota
+	disallowAsterisk
+)
```

Comment:
Using a bool was considered, but we opted to go for a const to make it clearer in functions where `parseIdentifier` is used. This way it is clear to see that the part before the `.` in a Go object cannot be an asterisk but the part after can. (It would be so nice to have a Haskell style two element type with named members but alas)

Reply:
Ah okay I understand, thanks